### PR TITLE
boards: add Myra SiP Baseboard

### PR DESCRIPTION
**Pull request by fkokosinski**
This PR:
* ~adds driver for the mb85rsm1t fram~
* adds support for a new SoC in Zephyr - Myra SiP
* adds a new target that is based on the Myra SiP - Myra SiP Baseboard

### CODE REVIEW

File: boards/antmicro/myra_sip_baseboard/doc/img/myra_sip_baseboard.jpg
Comment (nordicjm) [inline boards/antmicro/myra_sip_baseboard/doc/img/myra_sip_baseboard.jpg:1]: Convert to webp, then put through https://tinypng.com/

File: boards/antmicro/myra_sip_baseboard/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+- Temperature accuracy: ±1°C
+- Pressure accuracy: ±1 hPa
+- Operating temperature range: -40°C to 85°C
+- Pressure range: 300-1100 hPa
+- QWIIC connectors: For easy peripheral expansion.
+
+Supported Features
+------------------
+
+The Zephyr ``myra_sip_baseboard`` board configuration supports the following hardware features:
```
Comment (nordicjm) [inline boards/antmicro/myra_sip_baseboard/doc/index.rst:125]: *board target

File: soc/antmicro/myra/Kconfig.soc [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+# The Myra is technically a SiP (System-in-Package) that consists of
+# the STM32G491REI6 MCU and additional components like FRAM. So for
+# Antmicro Myra SiP Baseboard the STM32G491XX SoC is to be indicated as
+# the build target, but since the Myra SiP is what a user can actually
+# see on a board, using only STM32G491XX in the Zephyr build
+# infrastructure might be confusing. That's why in the top level of SoC
+# definitions (for user-configurable options in Kconfig, for example)
+# the Myra term is used and STM32G491XX underneath.
+config SOC_MYRA
```
Comment (nordicjm) [inline soc/antmicro/myra/Kconfig.soc:12]: I'm really not convinced with this, it's an stm32, it should be handled as an stm32, all those extra parts should be in the board dts

**Review by nordicjm — CHANGES_REQUESTED** [review]
More I think about it the more it is a no, this is an stm32 with supporting components, they go in the board dts, do not create a "soc sip", use the existing stm32 one. This is no different than e.g. rm1xx_dvk where there is an spi flash inside the module

**Comment by kgugala** [comment]
Hi @nordicjm, thanks for your comment.

> More I think about it the more it is a no, this is an stm32 with supporting components, they go in the board dts, do not create a "soc sip", use the existing stm32 one

This approach would mean that the shared configuration between board targets that share the same SiP would have to include duplicated code that would be otherwise provided by the SiP symbol, though.

Additionally, there already are other SIPs defined in a similar way in Zephyr, e.g.:
* `soc/nordic/nrf91/Kconfig.soc`:
  * `CONFIG_SOC_NRF9161_LACA`
* `soc/nordic/nrf92/Kconfig.soc`:
  * `CONFIG_SOC_NRF9280`
* `soc/espressif/esp32s2/Kconfig.soc`:
  * `CONFIG_SOC_ESP32S2_SOLO_N*` (4 types)
  * `CONFIG_SOC_ESP32S2_MINI*` (2 types)
  * `CONFIG_SOC_ESP32S2_WROOM`
  * `CONFIG_SOC_ESP32S2_WROVER*` (3 types)
* `soc/espressif/esp32s3/Kconfig.soc`:
  * `CONFIG_SOC_ESP32S3_MINI*` (2 types)
  * `CONFIG_SOC_ESP32S3_WROOM*` (9 types)

For now, all the SIPs in supported in Zephyr are in produced by the same vendor as the main SoC used in the SIP, however this is not always a case in the real world. Another example (similar to Myra) of such a SIP is https://octavosystems.com/octavo_products/osd32mp15x/ (this one is not yet supported in Zephyr, but we plan to add it soon)

**Comment by nordicjm** [comment]
The 91* and 92* are normal SoCs in tree so I don't see what you mean. The espressif ones are different chips with different internal configurations, this one you have here is just a specific stm32 chip with peripherals around it, there is no reason to do it as you've done it here. Not that SiP's aren't important and support for that in hwmv2 is planned for the future, but there is nothing groundbreaking here that cannot be done as every other "SiP" in zephyr is done with a normal SoC definition for the actual SoC and then anything inside of a "package" being done at board level

**Comment by fkokosinski** [comment]
> The 91* and 92* are normal SoCs in tree so I don't see what you mean

`SOC_NRF9161_LACA` is explicitly referred to as a SiP in a comment in Kconfig: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/nordic/nrf91/Kconfig.soc#L31-L38, and uses the same justification that we had provided:

> The nRF9161 is technically a SiP (System-in-Package) that consists of the nRF9120 SoC and additional components like PMIC, FEM, and XTAL, so for nrfx/MDK the nRF9120 SoC is to be indicated as the build target, but since the nRF9161 is what a user can actually see on a board, using only nRF9120 in the Zephyr build infrastructure might be confusing. That's why in the top level of SoC definitions (for user-configurable options in Kconfig, for example) the nRF9161 term is used and nRF9120 underneath.

The same is true for for `CONFIG_SOC_NRF9280`: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/nordic/nrf92/Kconfig.soc#L6-L10.

> The espressif ones are different chips with different internal configurations

Again, those SoCs that Karol had mention are listed under the "SiP" section comment in their respective Kconfig files https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/espressif/esp32s2/Kconfig.soc#L34:

> SiP with external flash / psram

**Comment by fkokosinski** [comment]
> The 91* and 92* are normal SoCs in tree so I don't see what you mean

Those nRFs mentioned by @kgugala are referred to as SiPs by comments in their respective Kconfigs. As for `CONFIG_SOC_NRF9161_LACA`, the comment is here: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/nordic/nrf91/Kconfig.soc#L31-L3

Plus, on the vendor's site nRF9161 is also explicitly called a SiP: https://www.nordicsemi.com/Products/nRF9161.

> The nRF9161 is technically a SiP (System-in-Package) that consists   of the nRF9120 SoC and additional components like PMIC, FEM, and XTAL, so for nrfx/MDK the nRF9120 SoC is to be indicated as the build target, but since the nRF9161 is what a user can actually see on a board, using only nRF9120 in the Zephyr build infrastructure might be confusing. That's why in the top level of SoC definitions (for user-configurable options in Kconfig, for example) the nRF9161 term is used and nRF9120 underneath.

And as for `CONFIG_SOC_NRF9280`, the comment is present here: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/nordic/nrf92/Kconfig.soc#L37

> The SiP selects the actual SoC complete with engineer revision and appropriate CPU

Additionally, this file also states the following: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/nordic/nrf92/Kconfig.soc#L12-L14

> Like it's done for the nRF91 family, let the nRF9280 SiP be represented by nRF9280 "SoC" in top-level SoC definitions and user-configurable Kconfigs, since that's what visible to users.

Which  means that those nRFs are explicitly defined as SiPs, and are definitely not "normal" SoCs.

---

> The espressif ones are different chips with different internal configurations

Again, those ESPs that Karol had mentioned are listed under the "SiP" section comment in their respective Kconfig files:
* https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/espressif/esp32s2/Kconfig.soc#L34
* https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/espressif/esp32s3/Kconfig.soc#L46

> SiP with flash and/or psram

---

>  but there is nothing groundbreaking here that cannot be done as every other "SiP" in zephyr is done with a normal SoC definition for the actual SoC and then anything inside of a "package" being done at board level

That's not exactly true, which the examples above prove. I.e. `CONFIG_SOC_NRF9161_LACA` (which is a Kconfig symbol for a SiP), selects `CONFIG_SOC_NRF9120`. I fail to see how nRF9161 is more "worthy" of being allowed this chain of Kconfig symbols as opposed to Myra, which other than having the STM32G491REI6 MCU in it, also integrates a 128kB FRAM chip, FTDI FT231XQ, crystal oscillators, LDOs etc. See more here: https://designer.antmicro.com/hardware/devices/myra-sip-interposer.

And here's the Kconfig snippet which demonstrates the Kconfig symbol chain: https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/nordic/nrf91/Kconfig.soc#L31-L41.

In summary: this PR doesn't really introduce any new way of defining a SiP; it just chooses the exact same convention that Zephyr had already opted to use.

File: boards/antmicro/myra_sip_baseboard/doc/img/myra_sip_baseboard.jpg
Comment (fkokosinski) [reply boards/antmicro/myra_sip_baseboard/doc/img/myra_sip_baseboard.jpg:1]: Done, filesize went down from 93k -> 34k

File: boards/antmicro/myra_sip_baseboard/doc/index.rst (hunk 1 above)
Comment (fkokosinski) [reply boards/antmicro/myra_sip_baseboard/doc/index.rst]: Fixed

**Review by henrikbrixandersen — DISMISSED** [review]
> adds driver for the mb85rsm1t fram
> adds support for a new SoC in Zephyr - Myra SiP
> adds a new target that is based on the Myra SiP - Myra SiP Baseboard

This needs to be split into at least two PRs: One for adding the Myra SiP support along with the baseboard, and another for adding the EEPROM driver (unless the EEPROM driver is needed for the board to boot, but it does not look like it is?).

**Comment by fkokosinski** [comment]
>  but it does not look like it is?

Correct - it's not needed for basic operation of the board. I've removed the driver from this PR.

**Review by kartben — CHANGES_REQUESTED** [review]

File: MAINTAINERS.yml [hunk 3]
```diff
@@ -126,6 +126,19 @@ ACPI:
   tests:
     - acpi
 
+Antmicro platforms:
+  status: maintained
+  maintainers:
+    - fkokosinski
+    - kgugala
+    - tgorochowik
```
[the remark this answers is not in the document]
Comment (fkokosinski) [reply MAINTAINERS.yml]: Done

File: boards/antmicro/myra_sip_baseboard/board.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+board:
+  name: myra_sip_baseboard
```
[the remark this answers is not in the document]
Comment (fkokosinski) [reply boards/antmicro/myra_sip_baseboard/board.yml:2]: Added full name

File: boards/antmicro/myra_sip_baseboard/doc/index.rst [hunk 5]
```diff
@@ -0,0 +1,266 @@
+.. _myra_sip_baseboard:
+
+Myra SiP Baseboard
+########################
```
[the remark this answers is not in the document]
Comment (fkokosinski) [reply boards/antmicro/myra_sip_baseboard/doc/index.rst]: the new directive is now used in the board doc

File: boards/antmicro/myra_sip_baseboard/doc/index.rst [hunk 6]
```diff
@@ -0,0 +1,266 @@
...
+- USB-C Connector for data and power
+- SHT45 temperature + humidity sensor
+- BME280 temperature + humidity + pressure sensor
+- QWIIC connectors for peripheral expansion
+- RTC with battery backup
+
+.. image:: img/myra_sip_baseboard.webp
+  :align: center
+  :alt: Myra SiP Baseboard
+
```
[the remark this answers is not in the document]
Comment (fkokosinski) [reply boards/antmicro/myra_sip_baseboard/doc/index.rst]: removed

File: boards/antmicro/myra_sip_baseboard/doc/index.rst [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+Serial Port
+-----------
+
+The Myra SiP Baseboard has 5 U(S)ARTs. The Zephyr console output is assigned to LPUART1. The default
+settings are 115200 8N1.
+
+Programming and Debugging
+*************************
+
+Applications for the ``myra_sip_baseboard`` board configuration can be built and flashed in the usual way (see :ref:`build_an_application` and :ref:`application_run` for more details).
```
[the remark this answers is not in the document]
Comment (fkokosinski) [reply boards/antmicro/myra_sip_baseboard/doc/index.rst]: fixed

File: boards/antmicro/myra_sip_baseboard/myra_sip_baseboard.yaml [hunk 8]
```diff
@@ -0,0 +1,28 @@
+identifier: myra_sip_baseboard
+name: Myra SiP Baseboard
+type: mcu
+arch: arm
+toolchain:
+  - zephyr
+  - gnuarmemb
+  - xtools
```
[the remark this answers is not in the document]
Comment (fkokosinski) [reply boards/antmicro/myra_sip_baseboard/myra_sip_baseboard.yaml]: removed `xtools` from the list

**Review by tejlmand — DISMISSED** [review]
Approving.

There's currently no clean concepts of SiPs in Zephyr, but HWMV2 has paved the way to improve this by offering a better and more scalable model.
But for now, SiPs are generally exposed as being a SoC.

I think the current suggestion of presenting the Myra SiP with it's own SoC name in the soc / board yaml and Kconfig is acceptable as the system is for now.
I also believe it will make it less intrusive when having a true SiP support, as the names / symbols in that case will not need to change (assumption :see_no_evil: ).

Having dedicated SiP Kconfig symbol selecting the true STM SOC also allows to different cases where it's important to distinguish the SiP from the SoC.

File: boards/antmicro/myra_sip_baseboard/myra_sip_baseboard_defconfig [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+CONFIG_SERIAL=y
+CONFIG_GPIO=y
+CONFIG_CLOCK_CONTROL=y
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+CONFIG_ARM_MPU=y
+CONFIG_HW_STACK_PROTECTION=y
+CONFIG_MAIN_STACK_SIZE=4096
```
Comment (nordicjm) [inline boards/antmicro/myra_sip_baseboard/myra_sip_baseboard_defconfig:11]: why is this needed?

File: boards/antmicro/myra_sip_baseboard/myra_sip_baseboard_defconfig [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+
+CONFIG_RTC=y
+CONFIG_RTC_STM32=y
+CONFIG_RTC_LOG_LEVEL_WRN=y
```
Comment (nordicjm) [inline boards/antmicro/myra_sip_baseboard/myra_sip_baseboard_defconfig:15]: why is RTC needed for every single build e.g. hello world?

**Review by nordicjm — DISMISSED** [review]

**Comment by fkokosinski** [comment]
Hey @MaureenHelm, could you take a look at this PR? Thanks!

**Review by kartben — CHANGES_REQUESTED** [review]
Spotted some issues with the board's doc page. Thanks!

**Review by kartben — APPROVED** [review]

**Review by tejlmand — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
